server.py

```diff
@@ -0,0 +1,352 @@
+        """
+        order_id = flask.request.json["order_id"]
+        delivery_person_id = flask.request.json["delivery_person_id"]
+        assign = fos.assign_deliver_person_to_deliver_order(order_id, delivery_person_id)
+        return json.dumps("{}".format(success if assign else failed))
+
+    @app.route('/employee/update_order', methods=['PUT'])
+    def update_order():
+        """
+        >> curl -H "Content-Type: application/json" -X PUT -d "{\"order_id\":9, \"order_status\":\"En route\"}" "localhost:8080/employee/update_order"
+        >> curl -H "Content-Type: application/json" -X PUT -d "{\"order_id\":9, \"order_status\":\"Delivered\"}" "localhost:8080/employee/update_order"
+        """
+        order_id = flask.request.json["order_id"]
+        order_status = flask.request.json["order_status"]
+        update = fos.update_order(order_id, order_status)
+        return json.dumps("{}".format(success if update else failed))
+
+    @app.route('/employee/view_sales_today', methods=['GET'])
+    def view_sales_today():
+        """
+        >> curl -H "Content-Type: application/json" -X GET -d "{\"order_status\":\"'Delivered'\"}" "localhost:8080/employee/view_sales_today"
+        >> curl -H "Content-Type: application/json" -X GET -d "{\"order_status\":\"'En route'\"}" "localhost:8080/employee/view_sales_today"
+        >> curl -H "Content-Type: application/json" -X GET -d "{\"order_status\":\"'Checkedout'\"}" "localhost:8080/employee/view_sales_today"
+        """
+        order_status = flask.request.json["order_status"]
+        sales_today = fos.view_sales_today(order_status)
+        result = []
+        for i in sales_today:
+            sales_dict = {
+                "customer_name": i.cust_name,
+                "order_id": i.order_id,
+                "order_status": i.order_status,
+                "bill_amount": i.bill_amount,
+                "date_time": i.checkout_time
+            }
+            result.append(sales_dict)
+        return json.dumps(result)
+
+    @app.route('/employee/sum_revenue_today', methods=['GET'])
+    def sum_revenue_today():
+        """
+        >> curl -H "Content-Type: application/json" -X GET -d "{\"order_status\":\"'Delivered'\"}" "localhost:8080/employee/sum_revenue_today"
+        >> curl -H "Content-Type: application/json" -X GET -d "{\"order_status\":\"'En route'\"}" "localhost:8080/employee/sum_revenue_today"
+        >> curl -H "Content-Type: application/json" -X GET -d "{\"order_status\":\"'Checkedout'\"}" "localhost:8080/employee/sum_revenue_today"
+        """
+        order_status = flask.request.json["order_status"]
+        sum_rev_today = fos.sum_revenue_today(order_status)
+        for sum in sum_rev_today:
+            revObj = sum[0]
+            rev_dict = {
+                "today_revenue": revObj
+            }
+            return json.dumps(rev_dict)
+        print("Not found")
+        return json.dumps(not_found), 404
```

Comment:
This logic seems a bit off. I don't think you need a for loop for this. Since you're returning a list, i think there would be at most one element in the list. so just directly pick the first element.

```python
        if sum_rev_today:
  
            # First element is the summed revenue obj
            summ = sum_rev_today[0]
            revObj = summ[0]
            rev_dict = {
                "today_revenue": revObj
            }
            return json.dumps(rev_dict)
        return json.dumps(not_found), 404
```

Reply:
changed made